### DISCUSSION TITLE
Is it better to put all the validations in validator or can also put in hoverprovider?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1707] [Q&A] [answered]
I found that I **can detect some validation errors in hover provider** too.
Is it the **best practice to put every validation** in **validator** or **can scatter around in different providers**?

Will there be a case that validator can't detect some errors but hover provider can?

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

> Is it the best practice to put every validation in validator or can scatter around in different providers?

I wouldn't recommend moving validations out of the validator. The hover LSP feature is very specific to actually hovering over text, while validations (diagnostics in LSP speech) also appear in other places. For example, diagnostics will appear in the problems tab of your IDE, whereas hovers won't appear there.
